### PR TITLE
Add MD backfill patching

### PR DESCRIPTION
**Pull request by smcclure17**
snapshot built with these changes: https://covid-projections-git-bump-snapshot-3021-1880-covidactnow.vercel.app/us/maryland-md/?s=30211880

### CODE REVIEW

File: libs/datasets/nytimes_anomalies.py [hunk 1]
```diff
@@ -155,6 +155,14 @@ def filter_by_nyt_anomalies(dataset: MultiRegionDataset) -> MultiRegionDataset:
     for index, row in applicable_nyt_anomalies.iterrows():
         if row[NYTimesFields.OMIT_FROM_ROLLING_AVERAGE] == "yes":
             location_id = row[CommonFields.LOCATION_ID]
+
+            # NOTE(sean) 03/08/22: ignoring MD anamolies in december 2021 to allow
```
Comment (smcclure17) [inline libs/datasets/nytimes_anomalies.py:159]: This feels hacky, but was the easiest way to ignore the required anomalies

**Review by mikelehen — APPROVED** [review]
Looks good! Thanks for figuring this out.

File: tests/libs/datasets/custom_patches_test.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            "iso1:us#iso2:us-md,2021-12-10,0\n"
+            "iso1:us#iso2:us-md,2021-12-11,0\n"
+            "iso1:us#iso2:us-md,2021-12-12,0\n"
+            "iso1:us#iso2:us-md,2021-12-13,0\n"
+            "iso1:us#iso2:us-md,2021-12-14,0\n"
+            "iso1:us#iso2:us-md,2021-12-16,100\n"
+            "iso1:us#iso2:us-md,2021-12-17,18\n"
+            "iso1:us#iso2:us-md,2021-12-18,19\n"
+            "iso1:us#iso2:us-ma,2021-12-12,30\n"
+            "iso1:us#iso2:us-ma,2021-12-13,30\n"
```
[the remark this answers is not in the document]
Comment (smcclure17) [reply tests/libs/datasets/custom_patches_test.py:28]: Oh, neat, yeah that's more concise. I think I'll leave this as is, but will keep in mind for the future if that's alright. 

I think those second rows are `MA` not `MD`, just to test that other states won't be removed--I did use two very similar looking states 😅

[the remark this answers is not in the document]
Comment (mikelehen) [reply tests/libs/datasets/custom_patches_test.py:28]: Oh, haha, I kept looking back and forth figuring I was missing something but kept overlooking it.  Makes sense, thanks!

**Pull request merged**